### PR TITLE
137-consent-prompt-for-video-embed-block

### LINKED ISSUE #137: Consent prompt for video embed block
Add consent prompt like on this [page on nododos.nl](https://www.nododos.nl/nl/impactvolle-klanten/green-caravan-fossielvrij-elektrisch-laden/):

button | prompt
--- | ---
image | image

No Dodos website is based on Head Start. This is the [related code change](https://github.com/No-Dodos/nododos-website/commit/d161414392a222d6f0be4d9ab1468dab8e4b3506).

### PR DESCRIPTION
**Pull request by luukbrauckmann**
# Changes

- Adds consent alert to VideoEmbedBlock

# Associated issue

Resolves #137 

# How to test

1. Open preview link
2. Scroll down to the Video
3. Click on the video
4. There should be an overlay popping up asking for permission
5. Click the allow button.
6. Video should start.
7. Check if this permission is added in the local storage.

# Checklist

- [x] I have performed a self-review of my own code
- [x] I have made sure that my PR is easy to review (not too big, includes comments)
- [x] I have made updated relevant documentation files (in project README, docs/, etc)
- [x] ~~I have added a decision log entry if the change affects the architecture or changes a significant technology~~
- [x] I have notified a reviewer

### CODE REVIEW

**Comment by luukbrauckmann** [comment]
I could pretty easily copy all the code from the nododos project, because it's pretty generic.

**Comment by luukbrauckmann** [comment]
Readme already contained an item about the consent alert

**Comment by jbmoelker** [comment]
👏👏. We should do the same for the Embed Block, but probably in a new PR :)

File: src/blocks/VideoEmbedBlock/VideoEmbedBlock.astro [hunk 1]
```diff
@@ -82,23 +97,56 @@ function getVideoUrl({ loop, mute, video }: { loop: boolean; mute: boolean; vide
...
         srcset={getImageSrcSet({ video })}
         style={{ aspectRatio }}
       />
-    </a>
-    <iframe hidden allowfullscreen allow="autoplay" title={ title } style={{ aspectRatio }}></iframe>
+    </PlayButton>
+
+    <div class="consent-alert" role="alert" hidden>
+      <p class="consent-alert__text">
+        {t('consent_message_service', { service: providerName })}
```
Comment (decrek) [inline src/blocks/VideoEmbedBlock/VideoEmbedBlock.astro:120]: How do we handle content changes? Do we use migrations for content or just for content model changes?

**Review by decrek — COMMENTED** [review]
Why do we store the user's choice in local storage and not in a cookie? If we would do server side rendering we could render/not render the cookie preferences. I understand that that doesnt make sense for the video embed, but for cookie cookie preferences in general I would like a solution that saves the preferences in a cookie

File: src/blocks/VideoEmbedBlock/VideoEmbedBlock.client.ts [hunk 2]
```diff
@@ -1,48 +1,120 @@
...
+      this.#consentAndPlay.bind(this)
+    );
+  }
+
+  #consentAndPlay() {
+    $consent.setKey(this.#provider, true);
+    this.play({ focus: true });
+  }
+
+  #hasConsent() {
```
[the remark this answers is not in the document]
Comment (jbmoelker) [reply src/blocks/VideoEmbedBlock/VideoEmbedBlock.client.ts:99]: Looked this up with Declan recently. We did spot some warning flags. So let's keep consent prompt for Vimeo as well.

**Review by jbmoelker — APPROVED** [review]

File: src/blocks/VideoEmbedBlock/VideoEmbedBlock.astro (hunk 1 above)
Comment (jbmoelker) [reply src/blocks/VideoEmbedBlock/VideoEmbedBlock.astro:120]: Is on the backlog, see #27. So for now, manually.

**Comment by jbmoelker** [comment]
> 👏👏. We should do the same for the Embed Block, but probably in a new PR :)

See #157

**Pull request merged**